### ISSUE TITLE
[webpack 2] Cannot `export default function name() {}`

### ISSUE THREAD

**Issue by nthtran** [webpack/webpack#2399]
Given this, which is valid ES6 export:

```js
export default function name() {
  return null;
}
 ```

Webpack generates:

```js
/* harmony default export */ exports["a"] = function name() {
  return null;
}
	
(function () {
  function tagSource(fn, localName) {
    // ...
  }

  tagSource(name, 'name');
})();
```

Which results in this error:

```
	})();
   ^
TypeError: (intermediate value)(intermediate value)(...) is not a function
```

However `export default` after declaring the function works:

```js
function name() {
  return null;
}

export default name;
```

Which generates:

```js
function name() {
  return null;
}

var _default = name;
/* harmony default export */ exports["a"] = _default;

(function () {
  function tagSource(fn, localName) {
  // ...
  }

  tagSource(name, 'name');
  tagSource(_default, 'default');
})();
```

**Comment by sokra** [comment]
yep, it's a bug

**Comment by nthtran** [comment]
It's caused by a missing semi-colon here:

```js
exports["a"] = function name() {
  return null;
} // missing semi-colon
```

and the function being undefined here:

```js
tagSource(name, 'name'); // `name` is undefined
```

If we can somehow generate something like this both of these issues will be fixed:

```js
function name() {
  return null;
}

exports["a"] = name;
```

**Comment by gaearon** [comment]
To be clear, `tagSource` seems to come from React Hot Loader 3.

**Comment by gaearon** [comment]
Fixed on our side in `react-hot-loader@3.0.0-beta.0`.

**Comment by nthtran** [comment]
Confirmed fixed. Thanks @gaearon!
